SmartDeviceLink/SDLLockScreenPresenter.m

```diff
@@ -23,23 +23,32 @@ - (void)present {
         return;
     }
 
-    [[self.class sdl_getCurrentViewController] presentViewController:self.viewController animated:YES completion:nil];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [[self.class sdl_getCurrentViewController] presentViewController:self.viewController animated:YES completion:nil];
+    });
 }
 
 - (void)dismiss {
     if (!self.viewController) {
         return;
     }
 
-    [self.viewController.presentingViewController dismissViewControllerAnimated:YES completion:nil];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.viewController.presentingViewController dismissViewControllerAnimated:YES completion:nil];
+    });
 }
 
 - (BOOL)presented {
     if (!self.viewController) {
         return NO;
     }
 
-    return (self.viewController.isViewLoaded && (self.viewController.view.window || self.viewController.isBeingPresented));
+    __block BOOL presented = NO;
+    dispatch_sync(dispatch_get_main_queue(), ^{
+        presented = (self.viewController.isViewLoaded && (self.viewController.view.window || self.viewController.isBeingPresented));
+    });
```

Comment:
Calling `dispatch_sync(dispatch_get_main_queue()` while on the main thread causes deadlock and the app crashes